src/data/FileAdapter.js

```diff
@@ -108,6 +108,35 @@ export default class {
 
     return game;
   }
+
+  async cancelGame(game) {
+
+    return this._lock(`game_${game.id}`, 'write', async () => {
+      let gameRefreshed = await this._readFile(`game_${game.id}`, null);
+      if (gameRefreshed.state.started) {
+        return false;
+      }
+      let playerIds = new Set(
+        gameRefreshed.state.teams.filter(t => t && !!t.playerId).map(t => t.playerId)
+      );
+
+      [...playerIds].forEach(playerId =>
```

Comment:
Be sure to map and await all promises to make sure these transactions are complete before releasing the lock (returning from the function.

Reply:
done